chain/src/Pos/Chain/Genesis/Config.hs

```diff
@@ -87,16 +87,22 @@ data StaticConfig
       -- !FilePath = Path to file where 'GenesisData' is stored. Must be
       -- in JSON, not necessary canonical.
       -- !(Hash Raw) = Hash of canonically encoded 'GenesisData'.
-    deriving (Eq, Show, Generic)
+    deriving (Eq, Show)
 
 instance ToJSON StaticConfig where
-    toEncoding (GCSrc gcsFile gcsHash) =
-        pairs . pairStr "src"
-            . pairs  $ pairStr "file"
-                (toEncoding gcsFile) <> pairStr "hash" (toEncoding gcsHash)
+    toJSON (GCSrc gcsFile gcsHash) =
+        object [ "src"    .= object [ "file" .= gcsFile
+                                    , "hash" .= gcsHash
+                                    ]
+               ]
+    toJSON (GCSpec value) = object ["spec" .= (toJSON value)]
 
-    toEncoding (GCSpec value)          =
-        genericToEncoding defaultOptions (GCSpec value)
+    toEncoding (GCSrc gcsFile gcsHash) =
+        pairs $ mconcat [ "src" `pair`
+                              (pairs $ mconcat [ "file" .= gcsFile
+                                               , "hash" .= gcsHash
+                                               ])                   ]
```

Comment:
`mconcat [x] == x`: this could be rewritten as:
```
        pairs $ "src" `pair`
                    (pairs $ mconcat [ "file" .= gcsFile
                                     , "hash" .= gcsHash
                                     ])
```